msteams-platform/bots/how-to/format-ai-bot-messages.md

```diff
@@ -0,0 +1,401 @@
+                        abstract: Utilities.snippet(citation.content, 500)
+                    }
+                } as ClientCitation;
+            });
+        }
+
+        // If there are citations, modify the content so that the sources are numbers instead of [doc1], [doc2], etc.
+        const contentText = !citations ? content : Utilities.formatCitationsResponse(content);
+
+        // If there are citations, filter out the citations unused in content.
+        const referencedCitations = citations ? Utilities.getUsedCitations(contentText, citations) : undefined;
+
+        await context.sendActivity({
+            type: ActivityTypes.Message,
+            text: contentText,
+            ...(isTeamsChannel ? { channelData: { feedbackLoopEnabled } } : {}),
+            entities: [
+                {
+                    type: 'https://schema.org/Message',
+                    '@type': 'Message',
+                    '@context': 'https://schema.org',
+                    additionalType: ['AIGeneratedContent'],
+                    ...(referencedCitations ? { citation: referencedCitations } : {})
+                }
+            ] as AIEntity[]
+        });
+
+        return '';
+    };
+}
+```
+
+For more information, see [Teams AI code sample](https://github.com/microsoft/teams-ai/blob/main/js/packages/teams-ai/src/actions/SayCommand.ts).
+
+If you're using Bot Framework SDK to build your bot, include `citation` under `entities` array. Here's a code snippet for your reference:
+
+```javascript
+ await context.sendActivity({
+         type: ActivityTypes.Message,
+         text: `Hey I'm a friendly AI bot. This message is generated through AI [1]`, // cite with [1],
+        entities: [
+          {
+            type: "https://schema.org/Message",
+            "@type": "Message",
+            "@context": "https://schema.org",
+            citation: [
+              {
+                "@type": "Claim",
+                position: 1, // Required. Should match the [1] in the text above
+                appearance: {
+                  "@type": "DigitalDocument",
+                  name: "AI bot", // Title
+                  url: "https://example.com/claim-1", // Hyperlink on the title
+                  usageInfo: {
+                    "@type": "CreativeWork",
+                    name: "Confidential \\ Contoso FTE", // Sensitivity title
+                    description: "Only accessible to Contoso FTE", // Sensitivity description
+                  },
+                },
+              },
+            ],
+          },
+        ],
+}
+```
+
+| Property | Type | Required | Description |
+|--|--|--|--|
+| `citation.appearance.name` | | Yes | Denotes the title of the content referenced |
+| `citation.position` | | Yes | Denotes the position of the citation number |
+| `citation` | | No | Denotes the citation number for the content referenced |
+| `citation.@type` | | No | |
+| `citation.appearance.@type` | | No | |
```

Comment:
Add 'encodingFormat' property as well for completeness

Reply:
Updated as suggested.